### ISSUE TITLE
add support for CONAN_ARCHS_BUILD

### ISSUE THREAD

**Issue by SSE4** [conan-io/conan-package-tools#151]
conan 1.0 added new settings os_build and arch_build (useful for build tools and installers).
would be nice to add support for them in conan package tools, like adding new environment variable CONAN_ARCHS_BUILD

/cc @lasote @memsharded

**Comment by lasote** [comment]
Could you elaborate the use case? Thanks!

**Comment by lasote** [comment]
Could you elaborate the use case? Thanks!

**Comment by SSE4** [comment]
use-case is to build arch_build=x86 on one travis job, and arch_build=x64 on another. currently it seems to be complicated to configure.

**Comment by lasote** [comment]
I see. we have to think about it, currently, the "arch_build" is not a setting that belongs to a build configuration by default, is auto-detected and used in the default profile. 

Could be this use case related with #152 ? What is the need for running a Travis build arch_build=x86 on one Travis job, and arch_build=x64 on another?

**Comment by SSE4** [comment]
@lasote yeah, related
reason is that we're splitting Docker images, so i386 images are building for arch=x86 and arch_build=x86, while regular images are building for arch=x86_64 and arch_build=x86_64. that avoids conflicts of system requirements, for install.

**Comment by lasote** [comment]
So, could it be solved then with the new CONAN_DOCKER_32_IMAGES? #152

**Comment by SSE4** [comment]
didn't know about this, will try, thanks

**Comment by lasote** [comment]
It is new. I released it yesterday ;)

**Comment by SSE4** [comment]
@lasote  seems like option has no effect, same image is used (without -i386 suffix)
probably due to that code:
`use_docker_32 = (build.settings.get("arch", "") == "x86" and self.docker_32_images)`
code expects arch settings, which is obviously missing (I am not using arch and using only arch_build)

**Comment by lasote** [comment]
Released 0.12.1, please try!

**Comment by db4** [comment]
`arch_build` support is still needed for Windows. I would like to write:
```python
    builder = ConanMultiPackager()
    builder.add({"arch_build": "x86"})
    builder.run()
```
that in turn would call `conan create . user/channel -s arch_build=x86` (currently `arch_build` setting is ignored by ` builder.add()`)

BTW, bincrafters have made it working somehow. `build.py` from `cygwin_installer` package:
```python
from bincrafters import build_template_installer
import os

if __name__ == "__main__":
    builder = build_template_installer.get_builder()
    builder.add({"arch_build": os.environ["CONAN_ARCHS"]}, {}, {}, {})
    builder.run()
```
